app/controllers/collections_controller.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+class CollectionsController < ApplicationController
+  include ModelControllerBehavior
+  self.form_class = CollectionForm
+  self.resource_class = Collection
```

Comment:
So I think these are already factory classes, basically, and you wouldn't gain anything by making it a lambda on top of that. There's nothing stopping you from defining a FormFactory which responds to `#new` and returns the appropriate instance.